lib/interjectable.rb

```diff
@@ -42,6 +65,24 @@ def inject(dependency, &default_block)
           instance_variable_set(ivar_name, instance_eval(&default_block))
         end
       end
+
+      setter = :"#{dependency}="
+      if instance_variable_defined?(INJECTED_METHODS_IVAR)
+        instance_variable_set(INJECTED_METHODS_IVAR,
+                              (instance_variable_get(INJECTED_METHODS_IVAR) + [dependency, setter]).uniq)
+      else
+        instance_variable_set(INJECTED_METHODS_IVAR, [INJECTED_METHODS, dependency, setter])
+      end
```

Comment:
yeah this would be much easier as something like. And then we could have the reader method add in the `:injected_methods` constant itself

```
if !instance_variable_defined?(INJECTED_METHODS_IVAR)
  instance_variable_set(INJECTED_METHODS_IVAR)
end
instance_variable_get(INJECTED_METHODS_IVAR).append([dependency, setter])
```